src/Http/Headers/src/SetCookieHeaderValue.cs

```diff
@@ -587,6 +587,16 @@ private static int GetSetCookieLength(StringSegment input, int startIndex, out S
                     maxAge = -maxAge;
                 }
 
+                // Check if maxAge would cause TimeSpan.FromSeconds to overflow
+                // TimeSpan.MaxValue.TotalSeconds is approximately 922337203685.4775
+                const long MaxTimeSpanSeconds = 922337203685L;
+                const long MinTimeSpanSeconds = -922337203685L;
+                if (maxAge > MaxTimeSpanSeconds || maxAge < MinTimeSpanSeconds)
```

Comment:
```suggestion
                if (maxAge is > MaxTimeSpanSeconds or < MinTimeSpanSeconds)
```
Once https://github.com/dotnet/roslyn/issues/60534 lands, we get a free optimization.